### PR TITLE
refactor(ollama): use httpx for Ollama API

### PR DESCRIPTION
**Pull request by sandangel**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

A follow-up PR to use httpx for Ollama 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/llms/ollama.py [hunk 1]
```diff
@@ -17,6 +18,14 @@
     MessageRole,
 )
 
+DEFAULT_REQUEST_TIMEOUT = 30
```
Comment (logan-markewich) [inline llama_index/llms/ollama.py:21]: Probably best to make this configurable?

**Review by logan-markewich — APPROVED** [review]
Makes sense, thanks for the cleanup + async!

**Comment by chukya** [comment]
This was not tested properly.

httpx.ReadTimeout: timed out occurs on the latest version of llama_index. `llama-index==0.10.4`

**Comment by logan-markewich** [comment]
@chukya even if you set request_timeout=3000.0 ? Tbh I think the default could be higher than 30

**Pull request merged**